core/src/main/scala/dagr/core/tasksystem/Schedulable.scala

```diff
@@ -89,4 +89,30 @@ trait FixedResources extends ScheduleWithEmptyDefaultResources {
 trait VariableResources extends ScheduleWithEmptyDefaultResources {
 
   override def applyResources(resources: ResourceSet): Unit = this.resourceSet = resources
+
+  /** A crude attempt at estimating the minimum cores needed by assuming an infinite amount of memory. */
+  def minCores(start: Cores = Cores(1), end: Cores = Cores(32), step: Cores = Cores(1)): Option[Cores] = {
+    (start.value to end.value by step.value).flatMap { cores =>
+      val availableResources = new ResourceSet(Cores(cores), Memory.infinite)
+      this.pickResources(availableResources = availableResources).map(_.cores)
+    }.headOption
+  }
+
+  /** A crude attempt at estimating the minimum memory needed by assuming an infinite amount of cores. */
+  def minMemory(start: Memory = Memory("256m"), end: Memory = Memory("64g"), step: Memory = Memory("256m")): Option[Memory] = {
+    (start.value to end.value by step.value).flatMap { memory =>
+      val availableResources = new ResourceSet(Cores.infinite, Memory(memory))
+      this.pickResources(availableResources = availableResources).map(_.memory)
+    }.headOption
+  }
+
+  /** A crude attempt at estimating the minimum amount of resources needed by this task.  First estimates the # of cores
+    * by assuming an infinite amount of memory, and then estimates the amount of memory assuming an infinite # of cores.
+    * If either does not yield a value, an infinite amount for that resource type is used.
+    */
+  def minCoresAndMemory(maxCores: Cores = Cores(32), maxMemory: Memory = Memory("64g")): ResourceSet = {
+    val cores  = minCores(end=maxCores).getOrElse(Cores.infinite)
+    val memory = minMemory(end=maxMemory).getOrElse(Memory.infinite)
+    new ResourceSet(cores, memory)
```

Comment:
Perhaps at least test that if you offer it `cores` and `memory` after computing them separately, that that's sufficient?